yarn-project/prover-client/src/mocks/test_context.ts

```diff
@@ -183,21 +182,36 @@ export class TestContext {
     );
   }
 
-  public async processPublicFunctionsWithMockExecutorImplementation(
+  private async processPublicFunctionsWithMockExecutorImplementation(
     txs: Tx[],
     maxTransactions: number,
     txHandler?: ProcessedTxHandler,
     txValidator?: TxValidator<ProcessedTx>,
     executorMock?: (
       stateManager: AvmPersistableStateManager,
-      execution: PublicExecutionRequest,
-      globalVariables: GlobalVariables,
+      executionRequest: PublicExecutionRequest,
       allocatedGas: Gas,
-      transactionFee?: Fr,
-    ) => Promise<PublicExecutionResult>,
+      transactionFee: Fr,
+      fnName: string,
+    ) => Promise<AvmFinalizedCallResult>,
   ) {
+    // Mock the internal private function. Borrowed from https://stackoverflow.com/a/71033167
+    const simulateInternal: jest.SpiedFunction<
+      (
+        stateManager: AvmPersistableStateManager,
+        executionResult: any,
+        allocatedGas: Gas,
+        transactionFee: any,
+        fnName: any,
+      ) => Promise<AvmFinalizedCallResult>
+    > = jest.spyOn(
+      this.publicTxSimulator as unknown as {
+        simulateEnqueuedCallInternal: PublicTxSimulator['simulateEnqueuedCallInternal'];
+      },
+      'simulateEnqueuedCallInternal',
+    );
     if (executorMock) {
-      this.publicExecutor.simulate.mockImplementation(executorMock);
+      simulateInternal.mockImplementation(executorMock);
     }
```

Comment:
The function is private, so I found this way to mock it. Should I just make the function public because we need to mock this in the prover client (and also in PublicTxSimulator tests)?

Reply:
No strong preference for now.